bin/carl.ml

```diff
@@ -544,10 +544,23 @@ module CLI = struct
     Arg.(
       value & opt (some request_conv) None & info [ "X"; "request" ] ~doc ~docv)
 
-  let cacert =
+  let cacertfile =
+    let cert_conv =
+      let parse s = Ok (Cert.Filepath s) in
+      Arg.conv ~docv:"method" (parse, Cert.pp_print_cert)
+    in
+    let doc = "CA certificate to verify peer against" in
+    let docv = "file" in
+    Arg.(value & opt (some cert_conv) None & info [ "cacert" ] ~doc ~docv)
+
+  let cacertpem = 
+    let cert_conv =
+      let parse s = Ok (Cert.Certpem s) in
+      Arg.conv ~docv:"method" (parse, Cert.pp_print_cert)
+    in
     let doc = "CA certificate to verify peer against" in
     let docv = "file" in
-    Arg.(value & opt (some string) None & info [ "cacert" ] ~doc ~docv)
+    Arg.(value & opt (some cert_conv) None & info [ "cacert" ] ~doc ~docv)
```

Comment:
are you sure these two options behave correctly by themselves, and differently than one another?

My mental model of cmdliner says only one of them will be parsed, as they're both `--cacert`